### ISSUE TITLE
Optimization doesn't seem to work in regression tasks

### ISSUE THREAD

**Issue by amin-sagar** [IBM/AutoPeptideML#57]
Hello. Sorry for the non-descriptive title.
I am trying some sample datasets from the [PeptideGeneralizationBenchmarks](https://github.com/IBM/PeptideGeneralizationBenchmarks) repo. I have tried a classification task ([nc-antiviral.csv](https://github.com/IBM/PeptideGeneralizationBenchmarks/blob/main/downstream_data/nc-antiviral.csv)) and a regression task ([nc-binding.csv](https://github.com/IBM/PeptideGeneralizationBenchmarks/blob/main/downstream_data/nc-binding.csv)).
The classification task runs as expected and I get the results as follows.
```
Training Partition: 0.5
Training Partition: 0.6
Training Partition: 0.7
Training Partition: 0.8
Training Partition: 0.9
Training Partition: random
Run evaluation
        mcc       acc        f1  f1_weighted  precision    recall     auroc threshold
0  0.498914  0.727273  0.796610     0.753573   0.979167  0.671429  0.807937       0.5
1  0.550687  0.727273  0.714286     0.724167   0.967742  0.566038  0.768733       0.6
2  0.579952  0.750000  0.731707     0.746108   0.967742  0.588235  0.780604       0.7
3  0.737468  0.852273  0.826667     0.848977   1.000000  0.704545  0.852273       0.8
4  0.807856  0.897727  0.873239     0.895628   1.000000  0.775000  0.887500       0.9
5  0.574778  0.752809  0.807018     0.771443   1.000000  0.676471  0.838235    random
```
However on running the regression task, the optimization doesn't seem to run properly with the models which are supposed to work decently according to the peptide generalization paper. The output is as follows.

```
Best trial: 12. Best value: 31.8865:  36%|███████████████████████▊                                          | 36/100 [06:49<12:07, 11.37s/it]
Training Partition: 0.4
Training Partition: 0.5
Training Partition: 0.6
Training Partition: 0.7
Training Partition: 0.8
Training Partition: 0.9
Training Partition: random
Run evaluation
         mse       mae  pcc  spcc threshold
0  36.472097  5.786732  NaN   NaN       0.4
1  36.604030  5.762576  NaN   NaN       0.5
2  36.703381  5.796067  NaN   NaN       0.6
3  36.213902  5.719318  NaN   NaN       0.7
4  35.970784  5.699493  NaN   NaN       0.8
5  34.195470  5.558572  NaN   NaN       0.9
6  32.738886  5.464571  NaN   NaN    random
```
My config file for the classification task is 
```
databases:
  dataset:
    feat_fields: SMILES
    label_field: labels
    path: ./nc-antiviral.csv
    verbose: false
outputdir: ./results
pipeline:
  aggregate: true
  elements:
  - pipe-smiles-input:
      aggregate: false
      elements:
      - filter-smiles: {}
      name: smiles-input
      verbose: false
  - pipe-seq-input:
      aggregate: false
      elements:
      - filter-smiles:
          keep_smiles: false
      - canonical-cleaner:
          substitution: G
      - sequence-to-smiles: {}
      name: seq-input
      verbose: false
  name: macromolecules_pipe
  verbose: false
representation:
  elements:
  - chemberta-2:
      average_pooling: true
      batch_size: 32
      device: cpu
      engine: lm
      model: chemberta-2
  - molformer-xl:
      average_pooling: true
      batch_size: 32
      device: cpu
      engine: lm
      model: molformer-xl
  - peptideclm:
      average_pooling: true
      batch_size: 32
      device: cpu
      engine: lm
      model: peptideclm
  - ecfp-counts-14:
      engine: fp
      fp: ecfp-count
      nbits: 2048
      radius: 7
  verbose: true
test:
  algorithm: ccpart
  filter: 0.185
  min_threshold: 0.1
  partitions: all
  sim_arguments:
    bits: 1024
    data_type: molecule
    field_name: SMILES
    min_threshold: 0.1
    radius: 4
    sim_function: jaccard
    verbose: 2
  threshold_step: 0.1
  verbose: 2
train:
  hspace:
    models:
      elements:
        lightgbm:
          colsample_bytree:
            log: false
            max: 1.0
            min: 0
            type: float
          learning_rate:
            log: true
            max: 0.1
            min: 1.0e-07
            type: float
          max_depth:
            log: false
            max: 10
            min: 3
            type: int
          min_split_gain:
            log: true
            max: 0.001
            min: 1.0e-10
            type: float
          n_estimators:
            log: false
            max: 1000
            min: 10
            type: int
          num_leaves:
            log: false
            max: 1024
            min: 8
            type: int
          reg_alpha:
            log: true
            max: 0.001
            min: 1.0e-10
            type: float
          subsample:
            log: false
            max: 1.0
            min: 0.5
            type: float
          verbose:
            type: fixed
            value: -1
        rf:
          ccp_alpha:
            log: true
            max: 0.001
            min: 1.0e-10
            type: float
          min_samples_split:
            log: false
            max: 100
            min: 2
            type: int
          n_estimators:
            log: false
            max: 1000
            min: 10
            type: int
      type: ensemble
    representations:
    - chemberta-2
    - molformer-xl
    - peptideclm
    - ecfp-counts-14
  optim_strategy:
    direction: maximize
    metric: mcc
    n_jobs: 16
    n_steps: 100
    partition: random
    patience: 20
    task: class
    trainer: optuna
  task: class
val:
  k: 10
  random_state: 1
  type: kfold
```
And my config file for the regression task is
```
databases:
  dataset:
    feat_fields: SMILES
    label_field: labels
    path: ./nc-binding.csv
    verbose: false
outputdir: ./results3
pipeline:
  aggregate: true
  elements:
  - pipe-smiles-input:
      aggregate: false
      elements:
      - filter-smiles: {}
      name: smiles-input
      verbose: false
  - pipe-seq-input:
      aggregate: false
      elements:
      - filter-smiles:
          keep_smiles: false
      - canonical-cleaner:
          substitution: G
      - sequence-to-smiles: {}
      name: seq-input
      verbose: false
  name: macromolecules_pipe
  verbose: false
representation:
  elements:
  - chemberta-2:
      average_pooling: true
      batch_size: 32
      device: cpu
      engine: lm
      model: chemberta-2
  - molformer-xl:
      average_pooling: true
      batch_size: 32
      device: cpu
      engine: lm
      model: molformer-xl
  - peptideclm:
      average_pooling: true
      batch_size: 32
      device: cpu
      engine: lm
      model: peptideclm
  - ecfp-counts-14:
      engine: fp
      fp: ecfp-count
      nbits: 2048
      radius: 7
  verbose: true
test:
  algorithm: ccpart
  filter: 0.185
  min_threshold: 0.1
  partitions: all
  sim_arguments:
    bits: 1024
    data_type: molecule
    field_name: SMILES
    min_threshold: 0.1
    radius: 4
    sim_function: jaccard
    verbose: 2
  threshold_step: 0.1
  verbose: 2
train:
  hspace:
    models:
      elements:
        lightgbm:
          colsample_bytree:
            log: false
            max: 1.0
            min: 0
            type: float
          learning_rate:
            log: true
            max: 0.1
            min: 1.0e-07
            type: float
          max_depth:
            log: false
            max: 10
            min: 3
            type: int
          min_split_gain:
            log: true
            max: 0.001
            min: 1.0e-10
            type: float
          n_estimators:
            log: false
            max: 1000
            min: 10
            type: int
          num_leaves:
            log: false
            max: 1024
            min: 8
            type: int
          reg_alpha:
            log: true
            max: 0.001
            min: 1.0e-10
            type: float
          subsample:
            log: false
            max: 1.0
            min: 0.5
            type: float
          verbose:
            type: fixed
            value: -1
        rf:
          ccp_alpha:
            log: true
            max: 0.001
            min: 1.0e-10
            type: float
          min_samples_split:
            log: false
            max: 100
            min: 2
            type: int
          n_estimators:
            log: false
            max: 1000
            min: 10
            type: int
      type: ensemble
    representations:
    - chemberta-2
    - molformer-xl
    - peptideclm
    - ecfp-counts-14
  optim_strategy:
    direction: minimize
    metric: mse
    n_jobs: 16
    n_steps: 100
    partition: random
    patience: 20
    task: reg
    trainer: optuna
  task: reg
val:
  k: 10
  random_state: 1
  type: kfold
```
Do you see anything wrong with my setup? I would again be really grateful for any suggestions.

Best,
Amin.

**Comment by RaulFD-creator** [comment]
I've found this problem this morning. Try installing autopeptideml again from the repo:

```bash
# First uninstall the current version in your environment
pip uninstall autopeptideml
# Then, reinstall from the repo
pip install git+https://github.com/IBM/AutoPeptideML
```

I've run your configuration file and I don't run into any issues. However, if the issue continues, please let me know and I'll try to dig into the issue. I would also recommend building the configuration file again, or just changing in `optim_strategy`, `direction` to `maximize` and `metric` to `pcc`.

Last few lines of the config file:

```yml
  optim_strategy:
    direction: minimize
    metric: mse
    n_jobs: 16
    n_steps: 100
    partition: random
    patience: 20
    task: reg
    trainer: optuna
  task: reg
val:
  k: 10
  random_state: 1
  type: kfold
```

**Comment by amin-sagar** [comment]
Thanks @RaulFD-creator .
This fixes the error.

Best,
Amin.
